template/src/core/util/object/objectUtils.ts

```diff
@@ -0,0 +1,29 @@
+/**
+ * Checks the given value if the value is object and not an array or null.
+ * @param {unknown} x The value to check.
+ * @returns {boolean} Returns `true` if `value` is an object and not an array or null, else returns `false`.
+ * @example
+ *
+ * isObject({})
+ * // => true
+ *
+ * isObject({a: "1"})
+ * // => true
+ *
+ * isObject(new Foo);
+ * // => true
+ *
+ * isObject([1, 2, 3])
+ * // => false
+ *
+ * isObject(Function)
+ * // => false
+ *
+ * isObject(null)
+ * // => false
+ */
+ function isObject(x: unknown): x is Record<string, any> {

```

Comment:
I think we agreed to rename this to `isRecord`. Even though these utils will come from the Aparat eventually, I think we should still rename it as they are added here now.
```suggestion
 function isRecord(x: unknown): x is Record<string, any> {

```